src/viam/sdk/services/mlmodel/private/proto.cpp

```diff
@@ -119,8 +124,60 @@ ::grpc::Status pb_value_to_tensor_t(const MLModelService::tensor_info& tensor_in
             };
 
         if (shape.size() != tensor_info.shape.size()) {
-            return {grpc::INTERNAL, make_error()};
+            // We will try to fix up a flat buffer when there should
```

Comment:
This mess is required until https://viam.atlassian.net/browse/RSDK-3096 is fixed, because even though it should be giving us something in the right format, it just hands us flat data.